packages/core/src/modules/credentials/protocol/v2/V2CredentialService.ts

```diff
@@ -942,22 +943,26 @@ export class V2CredentialService<CFs extends CredentialFormat[] = CredentialForm
       })
 
       // If any of the formats return false, we should not auto accept
+
       if (!shouldAutoRespondToFormat) return false
     }
 
     // not all formats use the proposal and preview, we only check if they're present on
     // either or both of the messages
-    if (proposalMessage.credentialPreview || offerMessage.credentialPreview) {
+    const noProposePreview =
+      !proposalMessage.credentialPreview || proposalMessage.credentialPreview.attributes.length === 0
+    const noOfferPreview = !offerMessage.credentialPreview || offerMessage.credentialPreview.attributes.length === 0
+
+    if (noProposePreview || noOfferPreview) {
```

Comment:
why is this change needed?

Reply:
Not sure what that is. I have taken it out and re-tested. Seems to be fine.